ambry-router/src/test/java/com.github.ambry.router/InMemoryRouter.java

```diff
@@ -331,9 +332,9 @@ public void run() {
     String operationResult = null;
     Exception exception = null;
     try {
-      String blobId = new BlobId(BlobId.DEFAULT_FLAG, ClusterMapUtils.UNKNOWN_DATACENTER_ID,
-          postData.getBlobProperties().getAccountId(), postData.getBlobProperties().getContainerId(),
-          getPartitionForPut()).getID();
+      String blobId = new BlobId(CommonTestUtils.getCurrentBlobIdVersion(), BlobId.BlobIdType.NATIVE,
+          ClusterMapUtils.UNKNOWN_DATACENTER_ID, postData.getBlobProperties().getAccountId(),
+          postData.getBlobProperties().getContainerId(), getPartitionForPut()).getID();
       if (blobs.containsKey(blobId)) {
         exception = new RouterException("Blob ID duplicate created.", RouterErrorCode.UnexpectedInternalError);
```

Comment:
I added some code in #792 to allow puts to InMemoryRouter with a specific blob ID version. I suppose that can be replaced by an `InMemoryRouter.setBlobIdVersion()` method.

Reply:
Haven't looked at that, how are you setting a specific version now? Will talk to you.

Reply:
I was using BlobIdV1, which overrides the getVersion method